web/src/components/storage/DriveEditor.tsx

```diff
@@ -21,40 +21,47 @@
  */
 
 import React from "react";
-import { Drive } from "~/types/storage/model";
-import { StorageDevice } from "~/types/storage";
-import DriveDeviceMenu from "~/components/storage/DriveDeviceMenu";
+import ConfigEditorItem from "~/components/storage/ConfigEditorItem";
 import PartitionableHeader from "~/components/storage/PartitionableHeader";
 import PartitionsMenu from "~/components/storage/PartitionsMenu";
 import SpacePolicyMenu from "~/components/storage/SpacePolicyMenu";
-import { Card, CardBody, CardHeader, CardTitle, Flex } from "@patternfly/react-core";
+import SearchedDeviceMenu from "~/components/storage/SearchedDeviceMenu";
+import { Drive } from "~/types/storage/model";
+import { model, StorageDevice } from "~/types/storage";
+import { useDeleteDrive } from "~/hooks/storage/drive";
 
-import spacingStyles from "@patternfly/react-styles/css/utilities/Spacing/spacing";
+type DriveDeviceMenuProps = {
+  drive: model.Drive;
+  selected: StorageDevice;
+};
 
-export type DriveEditorProps = { drive: Drive; driveDevice: StorageDevice };
+/**
+ * Internal component that renders generic actions available for a Drive device.
+ */
+const DriveDeviceMenu = ({ drive, selected }: DriveDeviceMenuProps) => {
+  const deleteDrive = useDeleteDrive();
+  const deleteFn = (device: model.Drive) => deleteDrive(device.name);
 
-const DriveHeader = ({ drive, driveDevice }: DriveEditorProps) => {
-  return (
-    <PartitionableHeader device={drive}>
-      <DriveDeviceMenu drive={drive} selected={driveDevice} />
-    </PartitionableHeader>
-  );
+  return <SearchedDeviceMenu modelDevice={drive} selected={selected} deleteFn={deleteFn} />;
 };
 
+export type DriveEditorProps = { drive: Drive; driveDevice: StorageDevice };
+
+/**
+ * Component responsible for displaying detailed information and available actions
+ * related to a specific Drive device within the storage ConfigEditor.
+ */
 export default function DriveEditor({ drive, driveDevice }: DriveEditorProps) {
   return (
-    <Card isCompact>
-      <CardHeader>
-        <CardTitle>
-          <DriveHeader drive={drive} driveDevice={driveDevice} />
-        </CardTitle>
-      </CardHeader>
-      <CardBody className={spacingStyles.plLg}>
-        <Flex direction={{ default: "column" }}>
-          <SpacePolicyMenu modelDevice={drive} device={driveDevice} />
+    <ConfigEditorItem
+      header={<PartitionableHeader drive={drive} device={driveDevice} />}
```

Comment:
This applies to all the "editors" in the storage area. I'm wondering if, now or maybe in a future iteration, it would be possible to simplify their props a bit, or at least make them a bit easier to follow.

To be honest, I’m still not fully confident I grasp the distinctions between things like `drive`, `device`, `selected`, etc., especially when jumping between components. That said, it could just be me still getting familiar with the storage business logic and maybe things are already in their best shape.

Just raising it in case there’s even a tiny bit of room for improvement 😊